# P07 - Aim 2: Examine microbial taxa and diversity on the chosen variable, Age, that significantly correlate to bacterial composition

## March 3rd, 2024

### Purpose:
To perform relative abundance plot, alpha and beta diversity metrics on the mouse familial dysautonomia dataset to determine the microbial dynamics across mouse age.

### Material: 
1. R & Rstudio
2. FD_rare_only_separate.Rdata (phylosesq object)

### Method 
#### relative abundance plot:
1. Import libraries: phyloseq, ggplot2, dplyr, reshape2, and tibble
2. Subset the phyloessq object by metadata categories -> FD_phyloseq_subset
3. Create a dataframe for the phyloseq subset
4. Convert absolute ASV reads into relative abundance for each sample in the otu table, and melt the dataframe 
5. Combine the tax and otu table by the ASV column
6. Subset the sample data with sample name and age only, then combine it with the combined tax and otu table by ASV column
7. Use ggplot to plot the relative abundance plot (x = sample name, y = relative abundance, color = phylum, stratified by age)

#### Alpha diversity (Chao1, Shannon):
1. In FD_phyloseq_subset, transform sample counts to integer
2. Melt the metric dataset
3. Plot ggplot (x = age, y = metric)

#### Beta diversity (Bray-Curtis, Jaccard):
1. Import libraries: phyloseq, ggplot2, dplyr, reshape2, vegan, and tibble
2. Subset the phyloessq object by metadata categories -> FD_phyloseq_subset
3. Factor the sample dataset to ensure the order of age
4. Create ordination for metric (parameter = PCoA, na.rm = T)
5. Plot the metric ordination (color = age)
6. Perform Permanova on each metric (dist.metric ~ Age,data=data)

### Output files:
#### Alpha diversity
1. Results stacked bar plot and boxplots
#### Beta diversity
2. Results PCoA plots
3. Permanova stats table

### Results: 

#### Alpha diversity

Relative abundance plot (phylum level)
> ![Relative abundance of phylum across mouse age](https://github.com/oliviakwon/MICB475_Team6/assets/158625508/132a18d3-9530-4235-a288-d0aeb0a0c697)

Boxplot - Chao-1
> <img src="/R_Project/Aim2/aim2_age_chao1.png" height="300">
Boxplot - Shannon
> <img src="/R_Project/Aim2/aim2_age_shannon.png" height="300">


#### Beta diversity
PCoA - Bray-Curtis 
> <img src="/R_Project/Aim2/aim2_age_bray.png" height="300">
PCoA - Jaccard 
> <img src="/R_Project/Aim2/aim2_age_jaccard.png" height="300">
Permanova - Bray-Curtis
> <img width="502" alt="Aim2 bray age adonis" src="https://github.com/oliviakwon/MICB475_Team6/assets/158625508/c78c34df-7bbf-4ccc-a2ae-6142a4cc60c7" height="180">
Permanova - Jaccard
> <img width="526" alt="Aim2 jaccard age adonis" src="https://github.com/oliviakwon/MICB475_Team6/assets/158625508/00e45e87-c925-47ec-9682-a06abaae0410" height="180">


### Discussion:

#### Alpha diversity 

##### Chao1:
*	The difference in number of species across that of the young, middle, and old mice was not significantly different.

##### Shannon:
*	Despite the median of the Shannon index increased from young, middle, to old mice, the difference in species distribution across that of the young, middle, and old mice was not significantly different. 


#### Beta diversity

##### Bray-Curtis:
*	22.8% of variance explained by axis 1 and 15.8% explained by axis 2.
*	The axis 1 variation likely to explain by the microbial composition difference between mice in middle and old age, while some young mice are more similar to the old or middle. 
*	In axis 2, samples from all ages are overlapping, hence age is not the factor to explain the variation in this axis.
*	The permanova result suggests that the microbial composition of mice significantly correlated with their age.
*	More specifically, post hoc test indicated that the composition dissimilarity between middle and young mice (padujst =0.006) or old and young mice (padujst = 0.012) was significant. 

##### Jaccard:
*	16.6% of variance explained by axis 1 and 11.8% explained by axis 2.
*	The axis 1 variation likely to explain by the difference in microbes present between mice in middle and old age, while some young mice are more similar to the old or middle. 
*	In axis 2, samples from mice at young and old are overlapping, while the presence/ absence of microbe shifts across axis 2.
*	The permanova result suggests that the presence of microbial members in mice significantly correlated with their age. 


## March 3rd, 2024

### Material: 

1. R & Rstudio
2. FD_rare_only_separate.Rdata (phylosesq object)

### Method 

#### Alpha diversity (Faith's Phylogenetic Diversity, Pielou's Evenness Index on Age):
1. In FD_phyloseq_subset, transform sample counts to integer
2. Melt the metric dataset
3. Plot ggplot (x = age, y = metric)
4. Run Kruskal-wallis tests on different metrics to determine significance 
5. Add Kruskal-wallis test results to the plot

#### Beta diversity (Unifrac Index, Weighted Unifrac Index on Age):
1. Import libraries: phyloseq, ggplot2, dplyr, reshape2, vegan, and tibble
2. Subset the phyloessq object by metadata categories -> FD_phyloseq_subset
3. Factor the sample dataset to ensure the order of age
4. Create ordination for metric (parameter = PCoA, na.rm = T)
5. Plot the metric ordination (color = age)
6. Perform Permanova on each metric (dist.metric ~ Age,data=data)
7. Perform post-hoc test for PERMANOVA with bonferroni method (code retrieved from [https://raunakms.github.io/diversity_cooking_fuel/06_01_permanova_test.html](https://raunakms.github.io/diversity_cooking_fuel/06_01_permanova_test.html))

### Output files:

#### Alpha diversity
* Box plots
* Kruskal-Wallis rank sum stats tables

#### Beta diversity
* Results PCoA plots
* Permanova stats tables
* Post-hoc stats tables

### Results: 

#### Alpha diversity 
* Shannon
    * Boxplot
      > <img src="/R_Project/Aim2/aim2_age_shannon.png" height="300">
    * Kruskal-Wallis test
      > <img src="/R_Project/Aim2/aim2_age_shannon_kw.png" height="60">
* Chao1
    * Boxplot
      > <img src="/R_Project/Aim2/aim2_age_chao1.png" height="300">
    * Kruskal-Wallis test
      > <img src="/R_Project/Aim2/aim2_age_chao1_kw.png" height="60">
* Pielou's evenness
    * Boxplot
      > <img src="/R_Project/Aim2/aim2_age_pielou.png" height="300">
    * Kruskal-Wallis test
      > <img src="/R_Project/Aim2/aim2_age_pielou_kw.png" height="60">
* Faith's phylogenetic diversity
    * Boxplot
      > <img src="/R_Project/Aim2/aim2_age_pd.png" height="300">
    * Kruskal-Wallis test
      > <img src="/R_Project/Aim2/aim2_age_pd_kw.png" height="60">

#### Beta diversity
* Bray
   * PCoA
      > <img src="/R_Project/Aim2/aim2_age_bray.png" height="300">
   * Permanova (pasted from above for easy comparison)
      > <img src="/R_Project/Aim2/aim2_age_bray_perm.png" height="180">
   * Post-hoc
      > <img src="/R_Project/Aim2/aim2_age_bray_posthoc.png" height="60">
* Jaccard
   * PCoA
      > <img src="/R_Project/Aim2/aim2_age_jaccard.png" height="300">
   * Permanova (pasted from above for easy comparison)
      > <img src="/R_Project/Aim2/aim2_age_jaccard_perm.png" height="180">
   * Post-hoc
      > <img src="/R_Project/Aim2/aim2_age_jaccard_posthoc.png" height="60">
* Unweighted unifrac
   * PCoA
      > <img src="/R_Project/Aim2/aim2_age_unifrac.png" height="300">
   * Permanova
      > <img src="/R_Project/Aim2/aim2_age_unifrac_perm.png" height="180">
   * Post-hoc
      > <img src="/R_Project/Aim2/aim2_age_unifrac_posthoc.png" height="60">
* Weighted unifrac
   * PCoA
      > <img src="/R_Project/Aim2/aim2_age_wunifrac.png" height="300">
   * Permanova
      > <img src="/R_Project/Aim2/aim2_age_wunifrac_perm.png" height="180">
   * Post-hoc
      > <img src="/R_Project/Aim2/aim2_age_wunifrac_posthoc.png" height="60">

### Discussion:

#### Alpha diversity 
* Shannon:
    * The p-value of 0.331 shows that the distribution of bacterial species across young, middle, and old mice was not significantly different in terms of abundance.
* Chao1:
    * The p-value of 0.3432 shows that the distribution of bacterial species across young, middle, and old mice was not significantly different in terms of species richness.
* Pielou's evenness
    * The p-value of 0.331 shows that the distribution of bacterial species across young, middle, and old mice was not significantly different in terms of evenness.
* Faith's phylogenetic diversity
    * The p-value of 0.4484 shows that the distribution of bacterial species across young, middle, and old mice was not significantly different in terms of richness.

#### Beta diversity
* Unweighted unifrac
   * 13.6% of variance is explained by axis 1 and 13.2% explained by axis 2.
   * Young, middle and old mice distribute differently along both axis 1 and axis 2. This suggests that each of the young, middle and old population hold different microbial species.
   * Permanova and post-hoc test result suggest species presence and abundance significantly correlate with mice age. 
* Weighted unifrac
   * 39.5% of variance is explained by axis 1 and 22.9% explained by axis 2.
   * The young- and middle- age mice samples overlap along both axes. This is likely the result of similar species presence and abundance between the young- and middle- age mice samples.
   * The old population overlapped with the young and middle age mice samples along Axis 2, but clustered at the left side of the young and middle age samples. This is likely the result of the dissimilarities between the old and the young-/middle- age mice samples.
   * Permanova and post-hoc test result suggest species presence and abundance significantly correlate with mice age.
 
## March 8th, 2024

### Material: 

1. R & Rstudio
2. FD_rare_only_separate.Rdata (phylosesq object)

### Method 
#### Beta diversity (Bray-Curtis on Sex and Genotype):
1. Import libraries: phyloseq, ggplot2, dplyr, reshape2, vegan, and tibble
2. Subset the phyloessq object by metadata categories -> FD_phyloseq_subset
3. Factor the sample dataset to ensure the order of sex and genotype
4. Create ordination for metric (parameter = PCoA, na.rm = T)
5. Plot the metric ordination (color = sex and color = genotype)
6. Perform Permanova on each metric 

### Output files:
#### Beta diversity
* Results PCoA plots
* Permanova stats tables

### Results:
#### Sex - Bray
* PCoA
   > <img src="/R_Project/Aim2/aim2_sex_bray.png" height="300">
* Permanova
   > <img src="/R_Project/Aim2/aim2_sex_bray_perm.png" height="180">
#### Genotype - Bray
* PCoA
   > <img src="/R_Project/Aim2/aim2_genotype_bray.png" height="300">
* Permanova
   > <img src="/R_Project/Aim2/aim2_genotype_bray_perm.png" height="180">
#### Weight - Bray
* PCoA
   > <img src="/R_Project/Aim2/aim2_weight_bray.png" height="300">
* Permanova
   > <img src="/R_Project/Aim2/aim2_weight_bray_perm.png" height="180">



### Discussion:
* (to be filled in)
#### Weight - Bray
*	22.8% of variance explained by axis 1 and 15.8% explained by axis 2.
*	Neither the axis 1 or 2 variation can be explained by the microbial composition difference by mouse weights.
*	However, the permanova result suggests that the microbial composition of mice significantly correlated with mouse weights.
*	This indicated that the variation that likely to be explained by mouse weights is not first or second axis of the PCoA but can be potentially explained by any one of the axes following axis 1 or 2 (eg., axis 3 or 4, etc).  
