# P09 - Aim 4: Predict metabolic pathways and compare their relative expression among the levels within each significant variable

 Mar 12, 2024

## Purpose:
To perform a functional analysis to predict what metabolic pathways are enriched or depleted in relation to the significant variable(s) linked to gut microbiome dysbiosis identified in aim 1

## Material: 
1. picrust2 qiime2 plugin
2. rep-seq.qza, table-no-mitochondria-no-chloroplast.qzv (from qiime2)
3. R & Rstudio

## Method:

#### Functional analysis using qiime2-picrust2 plugin on server:

1. Remove ASVs in table.qza with frequency < 5, generate feature-frequency-filtered-table.qza 
2. Run PICRUSt2 analysis (qiime2-picrust2 plugin) and output a directory that contains files with functional information

#### Downstream Analysis in RStudio (picrust2_aim2_metadata.R)

3. Convert pathway_abundance file generated under 'pathabun_exported' folder into human readable files (pathabun_feature-table.biom -> pathway_abundance.tsv)
4. In RStudio, load ggpicrust2 (version 1.7.3) package (34) and its dependencies
5. Load the pathway abundance table (pathway_abundance.tsv) generated from qiime2-picrust2 analysis, and 'FD_final_only_separate.Rdata' phyloseq object generated in previous aims. Derive sample_data of the phyloseq object and use as metadata. 
7. Remove NA's for the metadata column of interest, filter the abundance table to contain samples that also exist in the filtered metadata
8. Perform differential abundance analysis (DAA) using DESeq2, and annotate MetaCyc pathway to create an annotated dataframe
9. Filter p < 0.05 to keep only significant pathways. 
10. Use 'DESeq2_function.R' and run the DESeq2 function pathway abundance data and adjusting the metadata category of interest to be 'sex', to generate statistical information: log2FoldChange (log2FC), padjust, etc...
11. Generate a heatmap of pathways within both genotypes where |log2FC|>1, comparing between sex (female control, female FD, male control, male FD). Do this by combining 'sex' and 'genotype' into a new column 'group'. 
12. Generate a PCA plot of only mutant genotypes, that shows the distribution of male FD mice vs. female FD mice (no significance filtering).
13. Generate a volcano plot of only mutant genotypes, to show the most significant pathways (|log2FC|>2, padj < 0.05). 
14. Generate a bar plot of only mutant genotypes, representing log2FoldChange of each pathway (|log2FC|>2, padj < 0.05), where blue bars indicate male FD mice, and red bars indicate female FD mice. 

#### Correlation analysis between taxa composition information from Aim 2 and metabolic pathway abundance of FD mice ()

15. In RStudio, load tidyverse, phyloseq, ggsignif, picante, ggh4x, and their dependencies, 'FD_final_only_separate.Rdata', and 'pathway_abundance.tsv'.
16. Use tax_glom to calculate counts belonging to the same genus. Calculate relative abundance from abundance data.
17. Calulate relative abundance for each genus in each sample using taxa abundance information. Use Spearman’s rank correlation test to keep only genera in FD samples with padj<0.05. 
18. Keep only the 4 significant pathways determined in step 13 (|log2FC|>2, padj < 0.05): creatinine degradation II, catechol degradation to 2-oxopent-4-enoate II, catechol degradation II (meta-cleavage pathway), and methyl ketone biosynthesis.
19. Generate a scatterplot between the relative abundance of each FD sample of both sexes in the significant genus (g__Romboutsia) and the four significant pathways shown in step 13.

## Mar 20, 2024 (Not final plots)

* Heatmap (p < 0.05)
   > <img src="/R_Project/Aim4/pathway_heatmap.png"> 

* Volcano plot:
   > <img src="/R_Project/Aim4/volcano.png"> 
   
* PCA plot:
   > <img src="/R_Project/Aim4/pathay_pca.png"> 

* log2FC Bar plot:
   > <img src="/R_Project/Aim4/log2_bar.png"> 

## Mar 27, 2024 (Not final plots)

* Revised Heatmap (p < 0.05 and (Log2FoldChange > 1 or log2FoldChange < -1)), Female vs. Male for Control vs. FD

* Volcano plot (p < 0.05 and (Log2FoldChange > 1 or log2FoldChange < -1))
   > <img src="/R_Project/Aim4/FD_volcano.png"> 
* log2FC Bar plot (p < 0.05 and (Log2FoldChange > 1 or log2FoldChange < -1))
   > <img src="/R_Project/Aim4/FD_bar.png"> 

## Apr 3, 2024 (Final plots) 

* Final Heatmap (|Log2FC| > 1.5, padj < 0.05)
> <img src="/R_Project/Aim4/aim2_metadata_both_genotypes_log15_cutoff_heatmap.png"> 
* Final FD PCA plot 
> <img src="/R_Project/Aim4/aim2_metadata_mutant_pca.png"> 
* Final log2FC bar plot (|log2FC| > 2, padj < 0.05)
> <img src="/R_Project/Aim4/aim2_metadata_mutant_padjust_log2_cutoff_bar.png"> 
* Final scatterplot (correlation analysis)
> <img src="/R_Project/Aim4/g_Romboutsia_pathway_abundance.png"> 

## Code
* PICRUSt2 analysis using qiime2-picrust2 plugin: [qiime2-picrust2](PICRUSt2/PICRUSt2_code.txt)
* Downstream analysis of picrust2 analysis results (generating related plots): [picrust_r](/R_Project/Aim4/final_code/picrust_analysis_aim2_metadata.R)
* Correlation experiment analysis (generating scatter plot): [correlation_r](/R_Project/Aim4/final_code/tax-to-pathway-relative-abundance.r)

## Discussion:
1. Phylogenetic Investigation of Communities by Reconstruction of Unobserved States (PICRUSt2) analysis using qiime2-picrust2 revealed that sex was not the a main factor influencing functional diversity in FD mice. There was a significant overlap between the sexes, suggesting that there was no clear separation in the abundance of metabolic pathways between male and female FD mice. Although we observed clear taxonomic and diversity distinctions between male and female mice from our previous aims, our predictive functional anlysis indicated that this wasn't the case regarding metabolic or functional pathways.
2. However, through correlation experiment analysis, we observed that there was a strong correlation between the genus Romboutsia and the creatinine degradation pathway II exhibited a high correlation coefficient ρ (rho) = 0.96, p < 2.2e-16. 

## Future directions:
* In terms of the overall study objective, there was no strong evidence for distinctive metabolic pathway predictions between FD mice of different sexes. Since the dataset we worked with had notable difference between male and female mice samples, future directions could include studying a more even distribution of FD male and female mice and repeat the methods presented in this study to more accurately predict functional patterns in * gut FD male and female mice.
Since previous aims found notable taxonomic composition difference, further steps could include investigating how genetic and hormonal variations between sexes influence the gut microbiome. Since our study was done on mice model, comparative analyses of the gut microbiota and functions between FD mice and humans can further be performed to explore whether FD mouse models are a good representation of FD humans. 


